File: Longest Consectuive Sequence/step4-using-set.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set nums_set(nums.begin(), nums.end());
        int longest_consecutive_sequence = 0;

        for ( auto num: nums_set) {
            if ( !nums_set.contains(num - 1)) {
                int length = 1;
                while (nums_set.contains(num + length)) {
                    length++;
                }
                longest_consecutive_sequence = max(length, longest_consecutive_sequence);
            }
        }

        return longest_consecutive_sequence;
    }
};

File: Longest Consectuive Sequence/step4-using-set.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        longest_consecutive_sequence = 0

        for num in nums_set:
            if (num - 1) not in nums_set:
                length = 1
                while (num + length) in nums_set:
                    length += 1
                longest_consecutive_sequence = max(length, longest_consecutive_sequence)
        
        return longest_consecutive_sequence
